### PR TITLE
HotFix: speedup for extra long lines

### PR DESCRIPTION
**Pull request by babenek**
## Description

Please include a summary of the change and which is fixed.

- Fix speed issue for long line (45~50% speedup)
- Fix greedly regexes
- refuzzing

## How has this been tested?

Please describe the tests that you ran to verify your changes.

- [x] UnitTest
- [x] Benchmark

### CODE REVIEW

**Review by babenek — COMMENTED** [review]
TODO: rollback after https://github.com/Samsung/CredData/pull/143

File: credsweeper/credentials/line_data.py [hunk 1]
```diff
@@ -122,12 +147,29 @@ def clean_url_parameters(self) -> None:
...
+        if 3 > url_pos:
+            return
+        if not self.url_scheme_part_regex.match(line_before_value, pos=url_pos - 3, endpos=url_pos):
+            # check for scheme naming - must be matched
+            return
+        # use line only after ://
+        if self.url_chars_not_allowed_pattern.search(line_before_value, pos=url_pos + 3):
+            return
+        # all checks have passed - line before the value may be a URL
+        self.variable = self.variable.split('&')[-1].split('?')[-1].split(';')[-1]
```
Comment (xDizzix) [inline credsweeper/credentials/line_data.py:171]:
```suggestion
        self.variable = self.variable.rsplit('&', maxsplit=1)[-1].rsplit('?', maxsplit=1)[-1].rsplit(';', maxsplit=1)[-1]
```

File: credsweeper/credentials/line_data.py [hunk 2]
```diff
@@ -122,12 +147,29 @@ def clean_url_parameters(self) -> None:
...
+        self.value = self.value.split('&')[0].split(';')[0]
```
Comment (xDizzix) [inline credsweeper/credentials/line_data.py:172]:
```suggestion
        self.value = self.value.split('&', maxsplit=1)[0].split(';', maxsplit=1)[0]
```

**Review by xDizzix — APPROVED** [review]
+

**Pull request merged**